modules/gdpr_tasks/src/Plugin/QueueWorker/GdprTasksSarWorker.php

```diff
@@ -0,0 +1,423 @@
+   *   The task.
+   * @param bool $build_now
+   *   Whether to build the entity data immediate or defer to cron.
+   */
+  protected function initialise(TaskInterface $task, $build_now = FALSE) {
+    /* @var \Drupal\file\Plugin\Field\FieldType\FileFieldItemList $field */
+    $field = $task->get('sar_export');
+    /* @var \Drupal\Core\Field\FieldDefinitionInterface $field_definition */
+    $field_definition = $field->getFieldDefinition();
+    $settings = $field_definition->getSettings();
+
+    $config = [
+      'field_definition' => $field_definition,
+      'name' => $field->getName(),
+      'parent' => $field->getParent(),
+    ];
+    /* @var \Drupal\file\Plugin\Field\FieldType\FileItem $field_type */
+    $field_type = $this->fieldTypePluginManager->createInstance($field_definition->getType(), $config);
+
+    // Prepare destination.
+    $directory = $field_type->getUploadLocation();
+    file_prepare_directory($dirname, FILE_CREATE_DIRECTORY);
+
+    // Get a suitable namespace for gathering our files.
+    do {
+      // Generate a UUID.
+      $uuid = $this->uuid->generate();
+
+      // Check neither the file exists nor the directory.
+      if (file_exists("{$directory}/{$uuid}.zip") || file_exists("{$directory}/{$uuid}/")) {
+        continue;
+      }
+
+      // Generate the zip file to reserve our namespace.
+      $file = file_save_data('', "{$directory}/{$uuid}.zip", FILE_EXISTS_ERROR);
+    } while (!$file);
+
+    // Prepare the directory for our sub-files.
+    $content_directory = "{$directory}/{$uuid}";
+    file_prepare_directory($content_directory, FILE_CREATE_DIRECTORY | FILE_MODIFY_PERMISSIONS);
+
+    // Store the file against the task.
+    $values = [
+      'target_id' => $file->id(),
+      'display' => (int) $settings['display_default'],
+      'description' => '',
+    ];
+
+    $task->sar_export = $values;
+    $task->status = 'building';
+    $task->save();
+
+    // Start the build process.
+    if ($build_now) {
+      $this->build($task);
+    }
+    else {
+      // Queue for building.
+      $this->queue->createQueue();
+      $this->queue->createItem($task->id());
+    }
+
+  }
+
+  /**
+   * Build the export files.
+   *
+   * @param \Drupal\gdpr_tasks\Entity\TaskInterface $task
+   *   The task.
+   */
+  public function build(TaskInterface $task) {
+    /* @var \Drupal\file\Plugin\Field\FieldType\FileFieldItemList $field */
+    $field = $task->get('sar_export');
+    /* @var \Drupal\Core\Field\FieldDefinitionInterface $field_definition */
+    $field_definition = $field->getFieldDefinition();
+    $settings = $field_definition->getSettings();
+
+    $config = [
+      'field_definition' => $field_definition,
+      'name' => $field->getName(),
+      'parent' => $field->getParent(),
+    ];
+    /* @var \Drupal\file\Plugin\Field\FieldType\FileItem $field_type */
+    $field_type = $this->fieldTypePluginManager->createInstance($field_definition->getType(), $config);
```

Comment:
Ouch... this is painful... :( They should really put this on `FileFieldItemList`...

Reply:
Though we will definitely have a field item by this point on the `sar_export` field, so could just use that...